pkg/http-metrics/main.go

```diff
@@ -70,11 +106,31 @@ func main() {
 
 // ForwardRequest send all the request the the pod except for the ones having metrics/ in the path
 func ForwardRequest(res http.ResponseWriter, req *http.Request) {
+
+	klog.Infof("Received request %v", req)
+
 	requestTime := time.Now()
-	httputil.NewSingleHostReverseProxy(target).ServeHTTP(res, req)
-	responseTime := time.Now()
-	delta := responseTime.Sub(requestTime)
-	window.Append(float64(delta.Milliseconds()))
+
+	if proxy != nil {
```

Comment:
It's really unlikely (and for unlikely I mean almost impossible since you are starting first the proxy and then the server) that the proxy is nil but anyway, let's suppose for a moment that is `nil`. Why should we save metrics for a request that is not even forwarded because of a `nil` proxy?

Reply:
+1 agree

Reply:
Can you please do the opposite?
```
if proxy == nil {
    klog.Error("Proxy is nil")
    return
}
...
```
Always for the sake of readability 🥇